File: ch4/c/code4_6.c
#include <stdio.h>

int waysToScore(int n)
{
  if(n < 0) return 0;
  if(n == 0) return 1;
  return waysToScore(n - 10)
         + waysToScore(n - 5)
         + waysToScore(n - 3);
}

int main()
{
  printf("목적 점수를 입력하세요 : ");
  int N;
  scanf("%d", &N);
  printf("%d점 까지의 경우의 수는 %d 입니다.\n", N, waysToScore(N));
  return 0;
}

File: ch4/python/code4_6.py
def ways_to_score(n):
    if n < 0:
      return 0
    if n == 0:
        return 1
    return ways_to_score(n - 10) \
          + ways_to_score(n - 5) \
          + ways_to_score(n - 3)

N = int(input('목적 점수를 입력하세요 : '))
print('%d점까지의 경우의 수는 %d입니다.' % (N, ways_to_score(N)))